### ISSUE TITLE
quic-go v0.35.0

### ISSUE THREAD

**Issue by bt90** [caddyserver/caddy#5559]
https://github.com/quic-go/quic-go/releases/tag/v0.35.0

https://github.com/quic-go/quic-go/issues/3795 allows for a solid bump in HTTP/3 performance without having to deal with sysctl limits.

**Comment by francislavoie** [comment]
Yep! We have notifications for releases enabled too :sweat_smile: 

This one is a tricky upgrade unfortunately... we were relying on `quic.EarlyListener` being an interface to wrap it into a `fakeCloseQuicListener` so that we could reuse the listener across config reloads. Now that it's a struct though, the Go compiler isn't happy with us. We'll probably need to rewrite some things. :cry: /cc @marten-seemann

**Comment by francislavoie** [comment]
Oh nice, that was much simpler than I thought. Didn't know that type existed! Thanks!

**Comment by otbutz** [comment]
We also need to adjust our service definitions if we want to profit from https://github.com/quic-go/quic-go/issues/3795

https://github.com/caddyserver/dist/issues/97
